src/test/func/memory/memory.cc

```diff
@@ -313,7 +369,11 @@ void test_calloc_large_bug()
 int main(int argc, char** argv)
 {
   setup();
-
+#ifdef TEST_LIMITED
+  test_limited(8 * GiB);
```

Comment:
This test is asking for 8GiB of memory, but is not checking for 32bit?  Seems that this could be bad. Could you wrap the 8GiB and 2 GiB test to check for 64bit platform?  
```C++
if constexpr (bits::is64())
```

Reply:
It is actually setting the `RLIMIT_AS` to `8GiB`, the size to allocate the finally determined by `sysinfo.freeram` and upper-bounded by `2 GiB`. I was just using `8GiB` to provide a loose limit.

Reply:
So the first test shouldn't fail on 32bit, as it should do anything, and that test is passing. But the second one is failing. 
```
20/36 Test #23: func-memory-16 ...................***Exception: Child aborted  0.43 sec
limiting memory to 8388608 KiB
host freeram: 1559880 KiB
trying to alloc 194985 KiB
allocator initialised
success
...
limiting memory to 2097152 KiB
host freeram: 1558872 KiB
trying to alloc 194859 KiB
Out of memory
```
So it seems either the 2GiB test is failing, or tests are failing after 8GiB.  Could you order the tests in increasing size?  That would give us another piece of information.  Or even make it run them all and accumulate failures?  I believe this will be a QEMU issue as x86 32bit Linux is working.